framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -64,6 +64,23 @@ const std::unordered_set<std::string> kSurfaceExtensions = {
     VK_KHR_WIN32_SURFACE_EXTENSION_NAME,   VK_KHR_XCB_SURFACE_EXTENSION_NAME, VK_KHR_XLIB_SURFACE_EXTENSION_NAME
 };
 
+const std::unordered_set<VkColorSpaceKHR> kColorspaceSwapchainExtension = { VK_COLOR_SPACE_ADOBERGB_LINEAR_EXT,
```

Comment:
This is susceptible to code rot. Future extensions may add more enums that must be added but are very easy to forget to update this code.

I was going to suggest inverting this by only checking to see if the colorspace *isn't* VK_COLOR_SPACE_SRGB_NONLINEAR_KHR however, because there are multiple extensions that provide colorspaces, the code has to know which extension the colorspace comes from in order to check whether that extension is enabled.

The only reasonable alternative for me is generating a utility that maps a colorspace enum to its originating extension, which is effectively what is present here.

Generating something like the following:
```c++
std::unordered_map<VkColorSpaceKHR, const char*> kColorSpaceExtensionMap = { 
{VK_COLOR_SPACE_SRGB_NONLINEAR_KHR, nullptr}, // nullptr since its not from an extension
{VK_COLOR_SPACE_ADOBERGB_NONLINEAR_EXT, VK_EXT_SWAPCHAIN_COLOR_SPACE_EXTENSION_NAME},
{VK_COLOR_SPACE_BT2020_LINEAR_EXT, VK_EXT_SWAPCHAIN_COLOR_SPACE_EXTENSION_NAME},
...
```